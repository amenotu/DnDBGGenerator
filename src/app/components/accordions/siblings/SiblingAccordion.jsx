import React from 'react';
import { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Accordion, Icon, Segment } from 'semantic-ui-react';

function mapStateToProps(state, { id }) {
  return {
    birthOrder: state.getIn(['siblings', 'BirthOrder', id]),
    occupation: state.getIn(['siblings', 'Occupation', id]),
    alignment: state.getIn(['siblings', 'Alignment', id]),
    status: state.getIn(['siblings', 'Status', id]),
    attitude: state.getIn(['siblings', 'Attitude', id]),
    siblingClass: state.getIn(['siblings', 'Class', id]),
  };
}

class SiblingAccordion extends Component {
  static PropTypes = {
    id: PropTypes.Number,
    birthOrder: PropTypes.Object,
    occupation: PropTypes.Object,
    alignment: PropTypes.Object,
    status: PropTypes.Object,
    attitude: PropTypes.Object,
    siblingClass: PropTypes.Object,
  }

  constructor(props) {
    super(props);

    const { id } = this.props;

    this.state = {
      activeIndex: {
        0: id === 1 ? true : false,
      },
    };
  }

  componentWillReceiveProps(nextProps) {
    const { id } = this.props;
    this.setState({ activeIndex: {
        0: id === 1 ? true : false,
      },
    });
  }

  handleClick = (e, titleProps) => {
    const { activeIndex } = this.state;
    const index = titleProps;
    const newIndex = activeIndex[index] ? false : true;
    activeIndex[index] = newIndex;

    this.setState({ activeIndex });
  }

  render() {
    const {
      id,
      birthOrder,
      occupation,
      alignment,
      status,
      attitude,
      siblingClass,
    } = this.props;

    const { activeIndex } = this.state;

    const siblingOccupation = occupation === 'Adventurer' ? `${siblingClass.toLowerCase()} ${occupation.toLowerCase()}` : occupation.toLowerCase();

    return (
      <div>
        <Accordion inverted>
          <Accordion.Title
            active={ activeIndex[0] }
            onClick={ this.handleClick }>
            <Icon name='dropdown' />
            Sibling #{ id }
          </Accordion.Title>
          <Accordion.Content active={ activeIndex[0] }>
            <p>
              { `${birthOrder} Sibling: This sibling is a ${alignment.toLowerCase()} ${siblingOccupation}. They are ${status.toLowerCase()}. They feel ${attitude.toLowerCase()} towards you.` }
            </p>
          </Accordion.Content>
        </Accordion>
      </div>
    )
  };
}

export default connect(mapStateToProps, null)(SiblingAccordion);
