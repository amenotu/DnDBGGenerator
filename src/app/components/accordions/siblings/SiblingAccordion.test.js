import React from 'react';
import SiblingAccordion from './SiblingAccordion.jsx';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import renderer from 'react-test-renderer';
import Immutable from 'immutable';

const mockInitialState = Immutable.fromJS({
  siblings: {
    BirthOrder: {
      1: 'Younger',
    },
    Occupation: {
      1: 'Laborer',
    },
    Alignment: {
      1: 'Neutral',
    },
    Status: {
      1: 'Missing or unknown',
    },
    Attitude: {
      1: 'Friendly',
    },
    Class: {
      1: 'Rogue',
    }
  }
});

const configStore = configureStore();
const mockStore = configStore(mockInitialState);

const mockProps = { id: '1' };

describe('SiblingAccordion component', () => {
  it('renders correctly', () => {
    const renderedValue = renderer.create(<SiblingAccordion { ...mockProps } store={ mockStore } />).toJSON();
    expect(renderedValue).toMatchSnapshot();
  });

  it('sets state when clicked', () => {
    const wrapper = shallow(<SiblingAccordion { ...mockProps } store={ mockStore } />);
    wrapper.find({ 'data-id': 'siblingAccordion' }).simulate('click');
  });
});
