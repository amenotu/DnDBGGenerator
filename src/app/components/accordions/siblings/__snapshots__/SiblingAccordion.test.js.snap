// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SiblingAccordion component renders correctly 1`] = `
<div>
  <div
    className="ui inverted accordion"
  >
    <div
      className="title"
      onClick={[Function]}
    >
      <i
        aria-hidden="true"
        className="dropdown icon"
      />
      Sibling #
      1
    </div>
    <div
      className="content"
    >
      <p>
        Younger Sibling: This sibling is a neutral laborer. They are missing or unknown. They feel friendly towards you.
      </p>
    </div>
  </div>
</div>
`;
